### DISCUSSION TITLE
Understanding Filecoin with the goal of running a full mining node

### DISCUSSION THREAD

**Discussion by CCGLLC-knc** [filecoin-project/community#526] [Mining]
Hi all.

Introduction: I've been in crypto since 2014, having penned 6 SHA256 coins in the past, mostly for hire - it was quite educational.  Currently run a 300KW mostly Bitcoin datacenter.  At its peak, it had about 150 SHA256 miners, a nice GPU "whatever I wanted to play with GPU Coin" miner.  Currently (due to Bitcoin price) its down to a few S17 & S19 miners and a half dozen Innosilicon S11 dedicated to ScPrime.  Been with the ScPrime project for years, and was heavily involved in Storj some time ago (Heavily:  Ran over 2000 nodes).  Professionally functioned as an infrastructure technology directory (all storage, Unix, and Windows servers) for a Fortune 100 company prior to retiring into datacenter consulting for CEOs and CIOs until I finally retired for good 3 years ago.  

Just finished bringing a 10TB IPFS server online just for kicks (216.146.251.8:4001), provide 54TB to ScPrime's network, and run a Tor relay node.  Keeps me from getting bored <smile>.

Obviously, while researching IPFS, Filecoin came up.  I'd like to know more!  In the olden days, coins would have a basic stat page on bitcointalk.org that would provide basic stats like block time (apparently 30 seconds for Filecoin?), Max number of coins/tokens?  Current number of mined coins/tokens, reward method, pre-min amount for PoW coins,  etc.  Is there such a thing for Filecoin? 

Basically looking to see where I would be in the project if I invested enough to build a mining machine.

Also curious about if/how hardware requirements are expected to increase as the size of the blockchain increases.  e.g. I'd be pissed to spend $20K+ on a server and have it, or some component, become insufficient in 6 months.  Related, concerned that the current specs are close to the edge with regards to current technology.  For example, pcpartpicker does not even support the Ice Lake CPUs recommended (shame on them, but point is they were only released 2Q21).  Basically want to make sure that such a beast of a machine will continue to function for several years, without hitting something like a GPU memory restriction.

Given that most dual Xeon (my "goto" config for large servers) motherboards support 10 6gbyte/sec SATA connections, I could see having 160-200TB of storage along with a 2TB NVME disk for OS, ZFS cache, and 1TB for sealing cache (although I wonder if that space would be better served in the general ZFS log/cache space).  My understanding is that FIL needs to be purchase to functional "allow" storage space to be used.  How much FIL would I need to purchase to support, say, ~128TB of usable (10x16GB ZFS RaidZ2)?

Thanks in advance,

Kevin N. Carpenter

**Comment by TippyFlitsUK** [comment]
Hey Kevin 👋 

Welcome to the Filecoin community!

There are a few tools out there that will give you an array of stats on the Filecoin network and many more in active development. The [Filecoin Ecosystem](https://ecosystem.filecoin.io/) site is a great place to place to start to get an idea of what is currently out there.

Starboard Ventures has released [3 extremely useful Filecoin analytic products](https://www.starboard.ventures/solutions) that will answer many of your questions. There are also a few chain explorers available, [FilFox](https://filfox.info/en) is a popular choice. The [Filecoin Spec](https://spec.filecoin.io/) site is also a great place to visit to really dig into the network details.


With regards to your question about hardware requirements:

Filecoin storage providers use [lightweight snapshots](https://www.starboard.ventures/solutions) for interacting with the chain. If becoming an SP is your primary goal you don't need to worry about running the full chain. 

Generally speaking, resource requirements for running a Filecoin server have actually decreased since network launch in October 2020. Many aspects of the sealing process have been dramatically optimised by several multiples. Whilst a powerful system is still required, you will certainly not find yourself in a situation where your hardware is suddenly rendered redundant. I started my own SP operation a few months prior to network launch and am still using exactly the same components on my original server.

Regarding your last question about hardware configs you can see a few setups posted by existing storage providers [here](https://github.com/filecoin-project/lotus/discussions/6071). I would also thoroughly recommend joining [Filecoin Slack](https://filecoin.io/slack). You will find many discussions about architecture options and our community members are always happy to discuss new ideas!!

Welcome again to the community Kevin, I look forward to connecting with you again in the future!!

**Reply by CCGLLC-knc** [reply]
Thanks... I had previously viewed those existing storage provides, but became concerned when I saw questions posted like "How is that running for you" (one systems that appeared to be too small).

Tried joining Filecoin Slack with Google, it allows me to confirm my Google email, then tells me I need to use a different address <*sigh*>

Who do I need to contact to get a @starboardventures.io ID so that I can join the forum?

**Comment by crackerjack75** [comment]
how do i start mining crypto for beginners

**Comment by crackerjack75** [comment]
i got mineable miner cant seam how to use it
